docs/v1/P4Runtime-Spec.mdk

```diff
@@ -3293,10 +3312,73 @@ direct resources that are being queried.  Idle timeouts are not supported for
 static entries. If the table requires a priority value for entries, the server
 must populate the `priority` field appropriately, starting at 1 for the lowest
 priority entry and incrementing the value by 1 for each successive entry. Note
-that P4~16~ does not support assigning explicit priorities to static
-entries. When a priority value is required (&eg; for tables including `RANGE`,
-`TERNARY` or `OPTIONAL` matches), it is inferred based on the order in which
-entries appear in the table declaration.
+that P4~16~ does not support assigning explicit priorities to entries
+declared with `const entries`[^ConstEntriesPriorities].  When a
+priority value is required (&eg; for tables including `RANGE`,
+`TERNARY` or `OPTIONAL` matches), it is inferred based on the order in
+which entries appear in the table declaration.
+
+[^ConstEntriesPriorities]: This is not yet explicit in the P4~16~
+    language specification, but will become so if this or a similar
+    clarification is added to it:
+    <https://github.com/p4lang/p4-spec/pull/1259>
+
+### Preinitialized tables { #sec-preinitialized-tables }
+
+Preinitialized tables are those defined with an `entries` table
+property in the P4~16~ source code, with no `const` qualifier before
+`entries`, and at least one entry in that list.  In the P4Info, such
+tables have `has_initial_entries` flag equal to true, but
+`is_const_table` is false.  For tables declared with `const entries`,
+see Section [#sec-constant-tables].
+
+Every P4 table falls into one of three categories:
+
+* *Normal table*: Neither `entries` nor `const entries` are declared
+  in the source code, and thus `is_const_table` and
+  `has_initial_entries` will both be false.  A corner case is that if
+  it has `entries = { }` with no `const` before `entries`, &ie; an
+  empty list of entries, that is also a normal table.
```

Comment:
It occurs to me that handling this corner-case gracefully may be useful for cases where source code is dynamically-generated based on e.g. macros or other techniques. In such cases, certain "build profiles" may result in an empty `entries` list.